src/rvya25-profile.adoc

```diff
@@ -0,0 +1,466 @@
+   maximum of 64 bytes.
+- *Zihintpause* Pause hint.
+- *Zic64b* Cache blocks must be 64 bytes in size, naturally aligned in the
+address space.
+- *Zicbom* Cache-block management instructions.
+- *Zicbop* Cache-block prefetch instructions.
+- *Zicboz* Cache-Block Zero Instructions.
+- *Zfhmin* Half-precision floating-point.
+- *Zkt* Data-independent execution latency.
+- *V* Vector extension.
+- *Zvfhmin* Vector minimal half-precision floating-point.
+- *Zvbb* Vector basic bit-manipulation instructions.
+- *Zvkt* Vector data-independent execution latency.
+- *Zihintntl* Non-temporal locality hints.
+- *Zicond* Integer conditional operations.
+- *Zimop* may-be-operations.
+- *Zcmop* Compressed may-be-operations.
+- *Zcb* Additional compressed instructions.
+- *Zfa* Additional floating-Point instructions.
+- *Zawrs* Wait-on-reservation-set instructions.
+
+NOTE: The combination of "C" and "D" is incompatible with RVY as Zcd is an incompatible extension.
+
+Therefore the following mandatory extension is included instead:
+
+- *Zca* Compressed encodings excluding floating point loads and stores, including capability loads and stores for RVY
+
+The following mandatory extension is RVY specific:
+
+- *Zys* Sentry definition
+
+The following mandatory extension is in addition to RVA23U64:
+
+- *Zabhlrsc* Byte and halfword load-reserved and store conditional
+
+The following development option from RVA23U645 is mandatory in RVYA25U64:
+
+- *Zabha* Byte and halfword atomic memory operations.
+
+The following extension is mandatory in RVA23 but is _incompatible_ with RVY:
+
+- *Supm* Pointer masking, with the execution environment providing a means to
+   select PMLEN=0 and PMLEN=7 at minimum.
+
+NOTE: How to benefit from the combination of pointer masking and CHERI is a research topic.
+
+==== RVYA25U64 Optional Extensions
+
+===== Localized Options
+
+The following localized options are taken from RVA23U64:
+
+- *Zvkng* Vector crypto NIST algorithms with GCM.
+- *Zvksg* Vector crypto ShangMi algorithms with GCM.
+
+===== Development Options
+
+The following are new development options intended to become mandatory in a future RVA/RVYA profile.
+
+- *Zacas* Compare-and-Swap instructions.
+- *Ziccamoc* Main memory regions with both the cacheability and coherence PMAs
+  must provide `AMOCASQ` level PMA support.
+- *Zvbc* Vector carryless multiplication.
+- *Zama16b* Misaligned loads, stores, and AMOs to main memory regions that do not cross a naturally aligned 16-byte boundary are atomic.
+
+===== Expansion Options
+
+The following expansion options were also present in RVA22U64:
+
+- *Zfh* Scalar half-precision floating-point.
+- *Zbc* Scalar carryless multiply.
+- *Zvfh* Vector half-precision floating-point.
+- *Zfbfmin* Scalar BF16 converts.
+- *Zvfbfmin* Vector BF16 converts.
+- *Zvfbfwma* Vector BF16 widening mul-add.
+
+NOTE: *Zicfilp* Landing Pads, and *Zicfiss* Shadow Stack are redundant for CHERI, and so are currently listed as incompatible.
```

Comment:
As above. You want to support them for integer mode code, and Zicfilp has no incompatibility with CHERI. Yes, it may not be needed, but it isn't incompatible, and it can be complementary.

Reply:
Zicfiss you say integer only and doesn't exist for RVY.

Zicfilp I see no problem with allowing, especially given the labelling scheme has the ability to do some type enforcement that CHERI isn't designed to do, rather than just prevent JOP which we already do.

Reply:
OK - let's go with that for now.